test/modules/MetricsValidator/src/Program.cs

```diff
@@ -73,7 +79,9 @@ static async Task<int> MainAsync()
 
                             var result = new MethodResponse(Encoding.UTF8.GetBytes(testReporter.ReportResults()), (int)HttpStatusCode.OK);
 
+                            directMethodProcessingLock.Release();
```

Comment:
Should we add the Release() in a finally block to ensure it always gets released?

Reply:
oh good idea! Fixed.